src/status_im2/contexts/quo_preview/share/qr_code.cljs

```diff
@@ -2,22 +2,56 @@
   (:require [quo2.foundations.colors :as colors]
             [react-native.core :as rn]
             [quo2.core :as quo]
-            [status-im2.common.resources :as resources]))
+            [utils.image-server :as image-server]
+            [utils.re-frame :as rf]
+            [reagent.core :as reagent]
+            [status-im2.contexts.quo-preview.preview :as preview]))
+
+(def descriptor
+  [{:label "URL For QR"
+    :key   :text
+    :type  :text}
+   {:label   "Error Correction Level:"
+    :key     :error-correction-level
+    :type    :select
+    :options [{:key   :low
+               :value "Low"}
+              {:key   :medium
+               :value "Medium"}
+              {:key   :quart
+               :value "Quart"}
+              {:key   :highest
+               :value "Highest"}]}
+  ])
 
 (defn cool-preview
   []
-  (fn []
-    [rn/touchable-without-feedback {:on-press rn/dismiss-keyboard!}
-     [rn/view {:style {:padding-bottom 150}}
-      [rn/view {:style {:flex 1}}]
-      [rn/view
-       {:style {:padding-vertical 60
-                :flex-direction   :row
-                :justify-content  :center}}
-       [quo/qr-code
-        {:source (resources/get-mock-image :qr-code)
-         :height 250
-         :width  250}]]]]))
+  (let [state                  (reagent/atom {:text                   "https://status.im"
+                                              :error-correction-level :highest})
+        text                   (reagent/cursor state [:text])
+        error-correction-level (reagent/cursor state [:error-correction-level])
+        media-server-uri       (reagent/atom "")]
+    (fn []
+      (reset! media-server-uri (image-server/get-qr-image-uri-for-any-url
+                                {:url         @text
+                                 :port        (rf/sub [:mediaserver/port])
+                                 :error-level @error-correction-level
+                                 :qr-size     250}))
+      [rn/touchable-without-feedback {:on-press rn/dismiss-keyboard!}
+       [rn/view {:style {:padding-bottom 150}}
+        [preview/customizer state descriptor]
+        [rn/view {:style {:flex 1}}]
+        [rn/view
+         {:style {:padding-vertical 60
+                  :flex-direction   :row
+                  :justify-content  :center}}
+         [rn/view
+          [rn/text {:style {:padding 20 :flex-shrink 1}} "Media server url -> " @media-server-uri]
+          [quo/qr-code
+           {:source {:uri @media-server-uri}
+            :height 250
+            :width  250}]
+         ]]]])))
```

Comment:
There are some instances of dangling chars in this file ;)

Reply:
addressed here : https://github.com/status-im/status-mobile/pull/15755/commits/ddbe0e089796e1dead6cddb88adda90dc4acd8ba